### PR TITLE
Support automatical update expected in e2e test framework

### PR DESCRIPTION
**Pull request by mengqiy**
This PR make it convient to do bulk updates in expected directories like: https://github.com/GoogleContainerTools/kpt-functions-catalog/pull/181

### CODE REVIEW

**Review by natasha41575 — APPROVED** [review]
lgtm

File: pkg/test/runner/runner.go [hunk 1]
```diff
@@ -19,6 +19,11 @@ type Runner struct {
 }
 
 const (
+	// If this env is set to "true", this e2e test framework will update the
+	// expected diff and results if they already exist. If will not change
+	// config.yaml.
+	updateExpectedEnv string = "KPT_E2E_UPDATE_EXPECTED"
```
Comment (droot) [inline pkg/test/runner/runner.go:25]: I will be nice to have a full command somewhere (in README or a comment in Makefile) that a developer can copy and run to update the expected.

File: pkg/test/runner/runner.go [hunk 2]
```diff
@@ -290,3 +305,34 @@ func newExpected(path string) (expected, error) {
 
 	return e, nil
 }
+
+func updateExpected(tmpPkgPath, resultsPath, sourceOfTruthPath string) error {
```
Comment (droot) [inline pkg/test/runner/runner.go:309]: While using e2e, I have realized the diff format is not super friendly for debugging or expressing the output as human. It is optimized for the machine :)
What have been your experience.

**Review by droot — APPROVED** [review]
Some drive by comments.

File: pkg/test/runner/runner.go (hunk 1 above)
Comment (natasha41575) [reply pkg/test/runner/runner.go:25]: +1

File: pkg/test/runner/runner.go (hunk 2 above)
Comment (Shell32-Natsu) [reply pkg/test/runner/runner.go:311]: I will create PR to improve this. Will merge this one.

**Pull request merged**